android-components/components/browser/state/src/main/java/mozilla/components/browser/state/engine/middleware/TranslationsMiddleware.kt

```diff
@@ -41,6 +47,25 @@ class TranslationsMiddleware(
                     requestSupportedLanguages(context, action.tabId)
                 }
             }
+
+            is TranslationsAction.TranslateDataRequestedAction -> {
+                when (action.operation) {
+                    TranslationOperation.TRANSLATE -> {
+                        // No-op, not a data operation
+                    }
+                    TranslationOperation.RESTORE -> {
+                        // No-op, not a data operation
+                    }
+                    TranslationOperation.FETCH_LANGUAGES -> {
+                        // Bug 1877205
+                    }
+                    TranslationOperation.FETCH_PAGE_SETTINGS -> {
+                        scope.launch {
+                            requestTranslationPageSettings(context, action.tabId)
+                        }
+                    }
```

Comment:
You can combine the no-ops into a single expression using commas, which often are then placed last in the when:
```suggestion
                when (action.operation) {
                    TranslationOperation.FETCH_LANGUAGES -> {
                        // Bug 1877205
                    }
                    TranslationOperation.FETCH_PAGE_SETTINGS -> {
                        scope.launch {
                            requestTranslationPageSettings(context, action.tabId)
                        }
                    }
                    TranslationOperation.TRANSLATE,
                    TranslationOperation.RESTORE -> Unit
```

Reply:
Thanks, updated, didn't know that!